File: 2021/py/day1.py
with open("2021/_inputs/day1.txt", "r") as f:
    depths = [int(i) for i in f.read().splitlines()]

part1 = 0

for i, d in enumerate(depths):
    if i == 0:
        continue

    if d > depths[i - 1]:
        part1 += 1

sums = []
part2 = 0

for i in range(len(depths)):
    if i + 3 > len(depths):
        break
    sums.append(sum(depths[i : i + 3]))

for i, s in enumerate(sums):
    if i == 0:
        continue
    if s > sums[i - 1]:
        part2 += 1

print("Part 1:", part1)
print("Part 2:", part2)

File: 2015/c/day1.c
#include <stdio.h>
#include <string.h>

int main() {
	char input[65536];
	FILE *infile = fopen("2015/_inputs/day1.txt", "r");
	if (infile == NULL) {
		perror("Failed to open input file.");
		return 1;
	}

	fscanf(infile, "%s", &input);
	fclose(infile);

	int part1 = 1;
	int part2 = -1;

	for (int i = 0; i < strlen(input); i++) {
		if (input[i] == '(') part1++;
		else part1--;

		if (part2 == -1 && part1 < 0) part2 = i;
	}

	printf("Part 1: %d\n", part1);
	printf("Part 2: %d\n", part2);
}
